plugins/c8y_firmware_plugin/src/firmware_manager.rs

```diff
@@ -57,22 +60,23 @@ pub struct FirmwareManager {
     health_check_topics: TopicFilter,
     firmware_update_response_topics: TopicFilter,
     tedge_device_id: String,
-    http_client: Box<dyn C8YHttpProxy>,
+    download_req_sndr: Sender<DownloadRequest>,
+    download_res_rcvr: Receiver<DownloadResult>,
     local_http_host: String,
     persistent_dir: PathBuf,
-    tmp_dir: PathBuf,
     operation_timer: Timers<(String, String), ActiveOperationState>,
     timeout_sec: Duration,
+    req_in_progress: Option<(SmartRestFirmwareRequest, String)>,
```

Comment:
How do you ensure there is *at most* one running request?

Reply:
I was relying on the channel size of `1` to ensure that the second download request will block and hence will not overwrite this `req_in_progress`.

Reply:
Obsolete as I'm not using a tuple anymore.